core/dbt/parser/manifest.py

```diff
@@ -1579,14 +1605,19 @@ def _process_metric_node(
 def _process_metrics_for_node(
     manifest: Manifest,
     current_project: str,
-    node: Union[ManifestNode, Metric, Exposure],
+    node: Union[ManifestNode, Metric, Exposure, SavedQuery],
 ):
     """Given a manifest and a node in that manifest, process its metrics"""
 
+    metrics: List[List[str]]
     if isinstance(node, SeedNode):
         return
+    elif isinstance(node, SavedQuery):
+        metrics = [[metric] for metric in node.metrics]
```

Comment:
for my own understanding - why is this necessary? what does node.metrics look like?

Reply:
This necessary because the for loop that begins on 1620 expects each "metric" in the list of "metrics" to be a list of either 1 or 2 strings (i.e. `List[List[str]]`. Where the first string is the target_metric_name and the second (if it exists) is the `target_metric_package`. Saved query node metrics don't support package targeting. and it's `metrics` is just a list of strings (i.e. `List[str]`), with each string being a `target_metric_name`. Thus instead of handling them differently within the for loop that resolves the metrics, it was easier to just massage the saved query metrics to have the same shape.